abdiff/core/run_ab_transforms.py

```diff
@@ -1 +1,203 @@
+import os
+import time
+from datetime import timedelta
+from pathlib import Path
+from time import perf_counter
+
+import docker
+from docker.models.containers import Container
+
+from abdiff.config import Config, configure_logger
+from abdiff.core.exceptions import DockerContainersNotFoundError
+from abdiff.core.utils import read_run_json, update_or_create_run_json
+
+CONFIG = Config()
+logger = logging.getLogger(__name__)
+configure_logger(logger=logger, verbose=False)
+
+# init Docker client
+DockerClient = docker.from_env()
+
+
+def run_ab_transforms(
+    run_directory: str,
+    image_name_a: str,
+    image_name_b: str,
+    input_files: list[str],
+) -> tuple[list[str], ...]:
+    """Run Docker containers with versioned images of Transmogrifier."""
+    start_time = perf_counter()
+
+    # get run ID (used to label containers)
+    run_id = str(run_directory).split("/")[-1]
+
+    # create directories for 'transformed' outputs
+    transformed_directory_a, transformed_directory_b = create_directories(run_directory)
+
+    # create list to track container IDs for a given 'run'
+    container_ids = []
+
+    # create list of run configs
+    run_configs = [
+        (image_name_a, transformed_directory_a),
+        (image_name_b, transformed_directory_b),
+    ]
+
+    # transform input files using images A and B via Docker containers
+    # save transformed record JSON files in corresponding directory
+    for input_file in input_files:
+        source = (input_file.split("/")[-1]).split("-")[0]
+        for docker_image, transformed_directory in run_configs:
+            container = DockerClient.containers.run(
+                docker_image,
+                command=generate_transform_command(source, input_file),
+                detach=True,
+                environment={
+                    "AWS_ACCESS_KEY_ID": CONFIG.AWS_ACCESS_KEY_ID,
+                    "AWS_SECRET_ACCESS_KEY": CONFIG.AWS_SECRET_ACCESS_KEY,
+                    "AWS_SESSION_TOKEN": CONFIG.AWS_SESSION_TOKEN,
+                },
+                labels={
+                    "docker_image": docker_image,
+                    "source": source,
+                    "input_file": input_file,
+                    "run": run_id,
+                },
+                volumes=[f"{transformed_directory}:/tmp"],
+            )
+            logger.info(
+                f"Container '{container.id}' (Docker image: {docker_image}) "
+                f"RUNNING transform for '{source}' input_file: {input_file}."
+            )
+            container_ids.append(container.id)
+
+    # wait for all container runs to complete (status = "exited")
+    # get logs from all Docker containers
+    get_logs(run_id, container_ids, log_file=f"{run_directory}/transformed/logs.txt")
+
+    # clone run data and add list of input files
+    run_data = read_run_json(run_directory)
+    run_data.update({"input_files": input_files})
+    update_or_create_run_json(run_directory, run_data)
```

Comment:
Since you're just adding 1 key/value pair, I think you could just do `update_or_create_run_json(run_directory, {"input_files": input_files})` without the previous 2 lines